airflow/www/decorators.py

```diff
@@ -120,6 +120,7 @@ def wrapper(self, *args, **kwargs):
                                                                    dag_id)))):
                 return f(self, *args, **kwargs)
             else:
+                flash("DAG-level access is denied", "danger")
```

Comment:
how about the message change to `"User can't access {} DAG.".format(dag_id)` ?

Reply:
But when I implement the feature, the user can't even see the DAG from  the landing page(or can only see the DAG he can access. Is there something changed here?

Reply:
Sure. Let me update it.

Reply:
I don't know. But based on what I tested/observed now, users can see the DAGs to which that he/she doesn't have permissions.

May you check & confirm from your side as well?

Reply:
What I tested was to delete the two "all_dags" permissions from role "User", then log in as "User" role. I can still see all the DAGs (all are the "built-in" example DAGs though).

Reply:
@feng-tao , updated as advised. PTAL.

I think it's good to have this flash message, no matter if the user can or can not see the DAGs that he/she doesn't have permissions from the landing page (there may be cases that he/she is given a URL like `http://localhost:8080/tree?dag_id=<dag_to_check>` while he/she doesn't have access to `<dag_to_check>`).

Reply:
Yeah, I am afk and will merge tomorrow morning . And could you create a jira for that issue and assign it to me?

Reply:
Sure, will do that.

Reply:
> But when I implemented the feature, the user can't even see the DAGs that he doesn't have permissions from the landing page(or can only see the DAG he can access). Is there something changed here?

Created https://issues.apache.org/jira/browse/AIRFLOW-3949 to track the issue discussed above.